### DISCUSSION TITLE
[Tutorial] G-Research kaggle competition for crypto prediction

### DISCUSSION THREAD

**Discussion by rozeappletree** [ourownstory/neural_prophet#489] [Q&A - forecasting best practices] [answered]
I am new to this package and was interested to understand it better. To do so, I planning to test all the available features using data provided by [g-research crypto forecasting competition](https://www.kaggle.com/c/g-research-crypto-forecasting)

What will be the result of this PR.

> Series of notebooks featuring all available forecasting / feature engineering methods available.

What I need.

> - Help with unexpected behaviours
> - Help from some experienced people to know why forecasts are incorrect. If so, how can we improve it by making changes to the existing modules.

Timeline

> - I am planning to contribute for long term (minimum months to maximum years)
> - Will pickup action items based on your valuable suggestions

**Comment by rozeappletree** [comment]
I am starting with one step ahead predictions

**Comment by torronen** [comment]
Do you want to open a Discussion about price predictions? Kaggle seems to have a tight schedule, but if it is not a problem I think one option is to first try some commodity that may have better predictability due to cycles in nature. The Kaggle challenge is difficult because they are basically asking to disprove the random walk theory, and it might be impossible if random walk theory is correct.

**Comment by ourownstory** [comment]
Hi @INF800 Welcome to the community! 
I am happy to support you in this undertaking. Please feel free to reach out to me on Slack, too https://join.slack.com/t/neuralprophet/shared_invite/zt-sgme2rw3-3dCH3YJ_wgg01IXHoYaeCg
We can keep the discussion about features etc here on github (will convert this issue to a discussion)

However, as @torronen mentioned, stock markets, and mostly also crypto markets are not predictable as traditional time series. Because, when they are, someone will find out how, make money, and in doing so, remove the predictability.
Utra-high-frequency trading is still evolving more, though. 
Anyways, with clever integration of other data streams (e.g. google trends, twitter, monetary policies, etc) you may be able to find predictability in the past.

I look forward to hear of  your progress.

**Comment by gezabohus** [comment]
Hey guys,

I'm not sure if anyone is still watching this old topic; we'll see.

I am working with data with datapoints one per minute, with perhaps some points missing. The length of the data is maybe a month or so. The prediction should be the next (few) datapoint(s) or perhaps only if they are higher or lower than the last one seen.

Given that neural_prophet looks at yearly/weekly/daily seasonality, should I scale my timeline to one minute -> one day to take advantage of these features?

Thanks

**Reply by ourownstory** [reply, marked as answer]
Hi @gezabohus you can define a custom seasonality of any period!
here's some inspiration:
https://neuralprophet.com/how-to-guides/feature-guides/conditional_seasonality_peyton.html

**Reply by gezabohus** [reply]
@ourownstory, great, thanks for the help.
